model-server-openapi/specifications/model-server.yaml

```diff
@@ -13,481 +13,700 @@ info:
+      operationId: getRepositoryBranches
+      tags:
+        - branch management
       responses:
         "200":
-          $ref: '#/components/responses/200'
-        "500":
-          $ref: '#/components/responses/500'
-  /v2/repositories:
+          description: Successful listing of all repository branches
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/BranchList"
+        "404":
+          $ref: "#/components/responses/NoSuchRepository"
+        default:
+          $ref: "#/components/responses/Problem"
+
+    post:
+      operationId: createBranch
+      tags:
+        - branch management
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/Branch"
+      responses:
+        "201":
+          description: "Branch created successfully"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Branch"
+        "404":
+          description: "Repository not found or hash not found"
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/Problem"
+        "400":
+          description: "A branch with the given name already exists"
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/Problem"
+        default:
+          $ref: "#/components/responses/Problem"
+
+  /repositories/{repository}/branches/{branch}:
+    parameters:
+      - $ref: "#/components/parameters/RepositoryPath"
+      - $ref: "#/components/parameters/BranchPath"
+
     get:
-      operationId: getRepositories
+      operationId: getRepositoryBranch
+      tags:
+        - branch management
       responses:
         "200":
-          $ref: '#/components/responses/200'
-  /v2/repositories/{repository}/objects:
+          description: "Branch was found"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Branch"
+        "404":
+          description: Repository or branch does not exist.
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/Problem"
+        default:
+          $ref: "#/components/responses/Problem"
+
     put:
-      operationId: putRepositoryObjects
-      parameters:
-        - name: repository
-          in: "path"
-          required: true
-          schema:
-            type: string
+      operationId: updateBranch
+      summary: Updates an existing branch
+      description: Can be used to let an existing branch point to a new hash.
+      tags:
+        - branch management
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/Branch"
```

Comment:
`PUT /repositories/{repository}/branches/{branch}`: The request should not include `name`, only `current_hash`.

Reply:
I see, you thought about it by responing `400 Branch name cannot be changed`.
Is it common for APIs to always take and return the same object even, if the data can not be changed?

Reply:
The typical pattern would be to read and post/put the same type of resource. We can mark some of the properties as readOnly to avoid making them mutable but that doesn't account for all possible conditions.